pandas/core/groupby/generic.py

```diff
@@ -1863,15 +1864,144 @@ def _transform_general(self, func, engine, engine_kwargs, *args, **kwargs):
+                return self._transform_multiple_funcs(
+                    transformed_func,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+        else:
+            if isinstance(func, list):
+                func = maybe_mangle_lambdas(func)
+                return self._transform_multiple_funcs(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+            else:
+                return self._transform(
+                    func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
+                )
+
+    def _transform_multiple_funcs(
+        self,
+        func: Any,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> DataFrame:
+        results = []
+
+        if isinstance(func, dict):
+            for name, named_agg in func.items():
+                column_name = named_agg.column
+                agg_func = named_agg.aggfunc
+                result = self._transform_single_column(
+                    column_name,
+                    agg_func,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+                result.name = name
+                results.append(result)
+            output = concat(results, axis=1)
+        elif isinstance(func, list):
+            col_names = []
+            columns = (com.get_callable_name(f) or f for f in func)
+            func_pairs = zip(columns, func)
+            for idx, (name, func_item) in enumerate(func_pairs):
+                result = self._transform(
+                    func_item,
+                    *args,
+                    engine=engine,
+                    engine_kwargs=engine_kwargs,
+                    **kwargs,
+                )
+                results.append(result)
+                col_names.extend([(col, name) for col in result.columns])
+            output = concat(results, axis=1)
+            output.columns = MultiIndex.from_tuples(col_names)
+            output.sort_index(axis=1, level=[0], sort_remaining=False, inplace=True)
+
+        return output
+
+    def _transform_single_column(
+        self,
+        column_name: Hashable,
+        agg_func: Callable | str,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict | None = None,
+        **kwargs,
+    ) -> Series:
+        data = self._obj_with_exclusions[column_name]
+        groupings = self._grouper.groupings
+        result = data.groupby(groupings).transform(
+            agg_func, *args, engine=engine, engine_kwargs=engine_kwargs, **kwargs
         )
+        return result
+
+    @staticmethod
+    def _named_agg_to_dict(*named_aggs: tuple[str, NamedAgg]) -> dict[str, NamedAgg]:
+        valid_items = [
```

Comment:
Could you inline this function since it's only use once in this file?

Reply:
Yeah, simple is best, will do that.